templates/helpers.js

```diff
@@ -164,12 +169,56 @@ module.exports = function(docMap, options, getCurrent, helpers, OtherHandlebars)
             return docObject.package.repository.github || docObject.package.repository.url || docObject.package.repository;
           }
         },
-        getDocumentTitle: function(docObject){
+        getDocumentTitle: function(docObject) {
             var title = docMapInfo.getTitle(docObject) || 'CanJS';
             if (docObject.name === 'canjs' || (title && title.toLowerCase() === 'canjs')) {
                 return title;
             }
-            return 'CanJS - ' + title;
+
+            if (docObject.type !== "page") {
+                var group;
+                if (docObject.type === "module") {
+                    group = docMap[this.parent];
+                } else {
+                    var parentModule = docMap[docObject.parent];
+                    while(parentModule && parentModule.type !== "module") {
+                        parentModule = docMap[parentModule.parent];
+                    }
+                    if (parentModule) {
+                        group = docMap[parentModule.parent];
+                        title += " | " + (parentModule.title || parentModule.name);
+                    }
+                }
+                if(group) {
+                    var groupParent =  docMap[group.parent];
+                    if (groupParent) {
+                        if (groupParent.type === "module") {
+                            title += " | "  + (groupParent.title || groupParent.name);
+                            if (docMap[groupParent.parent].type === "group") {
+                                var groupGrandParent = docMap[groupParent.parent];
+                                var groupGrandParentParent = docMap[groupGrandParent.parent];
+                                if (!groupGrandParentParent) {
+                                    // the "API Docs" as parent is missing in some pages, can-connect
```

Comment:
Is this working around an issue that should be fixed in the docs?

Reply:
Yeah I think so.